ignore exec tests when passing DOCKER_EXECDRIVER lxc

jessfraz opened the pull request.

jessfraz committed:
Move InspectExecID test to exec.

Docker-DCO-1.1-Signed-off-by: Jessica Frazelle <jess@docker.com> (github: jfrazelle)

jessfraz committed:
Add build flag to exec test.

Docker-DCO-1.1-Signed-off-by: Jessica Frazelle <jess@docker.com> (github: jfrazelle)

jessfraz committed:
Update project/make.sh to use execdriver buildtag if passed.

Docker-DCO-1.1-Signed-off-by: Jessica Frazelle <jess@docker.com> (github: jfrazelle)

tianon commented on project/make.sh:
```diff
@@ -95,6 +95,10 @@ if [ -z "$DOCKER_CLIENTONLY" ]; then
 fi
 
+if [ "$DOCKER_EXECDRIVER" == "lxc" ]; then
+	DOCKER_BUILDTAGS+=" test_no_exec"
```
Single equals, and single quotes around `lxc` and ` test_no_exec`. :wink:

jfrazelle replied:
updated!

jessfraz committed:
Move links exec test & exec dir test.

Docker-DCO-1.1-Signed-off-by: Jessica Frazelle <jess@docker.com> (github: jfrazelle)

jfrazelle commented:
@ashahab-altiscale this will make sure any tests around exec won't be run on lxc, get rid of a bunch of errors, let me know if you think I missed any

ashahab-altiscale commented:
@jfrazelle This is awesome.
When I run my build, I see the following failures. Are these all accounted for?:

--- FAIL: TestExecApiCreateNoCmd (1.42s)
--- FAIL: TestExec (1.26s)
--- FAIL: TestExecInteractiveStdinClose (11.67s)
--- FAIL: TestExecInteractive (1.36s)
--- FAIL: TestExecAfterContainerRestart (2.51s)
--- FAIL: TestExecAfterDaemonRestart (4.49s)
--- FAIL: TestExecEnv (0.70s)
--- FAIL: TestExecExitStatus (1.30s)
--- FAIL: TestExecPausedContainer (2.29s)
--- FAIL: TestExecTtyCloseStdin (2.31s)
--- FAIL: TestExecTtyWithoutStdin (1.83s)

jfrazelle commented:
Yep yep :)

On Monday, January 19, 2015, Abin Shahab <notifications@github.com> wrote:

> @jfrazelle <https://github.com/jfrazelle> This is awesome.
> When I run my build, I see the following failures. Are these all accounted
> for?:
>
> --- FAIL: TestExecApiCreateNoCmd (1.42s)
> --- FAIL: TestExec (1.26s)
> --- FAIL: TestExecInteractiveStdinClose (11.67s)
> --- FAIL: TestExecInteractive (1.36s)
> --- FAIL: TestExecAfterContainerRestart (2.51s)
> --- FAIL: TestExecAfterDaemonRestart (4.49s)
> --- FAIL: TestExecEnv (0.70s)
> --- FAIL: TestExecExitStatus (1.30s)
> --- FAIL: TestExecPausedContainer (2.29s)
> --- FAIL: TestExecTtyCloseStdin (2.31s)
> --- FAIL: TestExecTtyWithoutStdin (1.83s)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/10194#issuecomment-70606248>.
>

ashahab-altiscale commented:
In that case, LGTM.

On Mon, Jan 19, 2015 at 9:35 PM, Jessie Frazelle <notifications@github.com>
wrote:

> Yep yep :)
>
> On Monday, January 19, 2015, Abin Shahab <notifications@github.com>
> wrote:
>
> > @jfrazelle <https://github.com/jfrazelle> This is awesome.
> > When I run my build, I see the following failures. Are these all
> accounted
> > for?:
> >
> > --- FAIL: TestExecApiCreateNoCmd (1.42s)
> > --- FAIL: TestExec (1.26s)
> > --- FAIL: TestExecInteractiveStdinClose (11.67s)
> > --- FAIL: TestExecInteractive (1.36s)
> > --- FAIL: TestExecAfterContainerRestart (2.51s)
> > --- FAIL: TestExecAfterDaemonRestart (4.49s)
> > --- FAIL: TestExecEnv (0.70s)
> > --- FAIL: TestExecExitStatus (1.30s)
> > --- FAIL: TestExecPausedContainer (2.29s)
> > --- FAIL: TestExecTtyCloseStdin (2.31s)
> > --- FAIL: TestExecTtyWithoutStdin (1.83s)
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/docker/docker/pull/10194#issuecomment-70606248>.
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/10194#issuecomment-70606834>.
>

jfrazelle commented:
ping @tianon :angel:

tianon commented:
LGTM

ashahab-altiscale commented:
Can this me merged? @crosbymichael ? At some point I intend to implement exec, but right now it would be nice to ignore these tests for lxc.

crosbymichael commented:
@jfrazelle needs a rebase

jfrazelle commented:
rebased!

ashahab-altiscale commented:
Is this test also an Exec test: TestRunMutableNetworkFiles? For me it fails on this line: https://github.com/docker/docker/blob/master/integration-cli/docker_cli_run_test.go#L2216

jfrazelle commented:
op yep I missed one I can fix it, thanks for finding that :)
